### PR TITLE
Add tutorial about inductor caching

### PR DESCRIPTION
**Pull request by oulgen**
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by svekars — APPROVED** [review]

**Comment by svekars** [comment]
Approved from the publishing perspective but someone with technical expertise should approve as well.

File: recipes_source/torch_compile_caching_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+Compile Time Caching in ``torch.compile``
+=========================================================
+**Authors:** `Oguz Ulgen <https://github.com/oulgen>`_ and `Sam Larsen <https://github.com/masnesral>`_
+
+Introduction
+------------------
+
+PyTorch Inductor implements several caches to reduce compilation latency. These caches are transparent to the user.
```
Comment (masnesral) [inline recipes_source/torch_compile_caching_tutorial.rst:8]: nit: Is "These caches are transparent to the user" relevant here? I'd just remove it.

File: recipes_source/torch_compile_caching_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+----------------------------
+
+Most of these caches are in-memory, only used within the same process, and are transparent to the user. An exception is the FX graph cache that stores compiled FX graphs. This cache allows Inductor to avoid recompilation across process boundaries when it encounters the same graph with the same Tensor input shapes (and the same configuration). The default implementation stores compiled artifacts in the system temp directory. An optional feature also supports sharing those artifacts within a cluster by storing them in a Redis database.
+
+There are a few settings relevant to caching and to FX graph caching in particular.
+The settings are accessible via environment variables listed below or can be hard-coded in Inductor’s config file.
+
+TORCHINDUCTOR_FX_GRAPH_CACHE
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This setting enables the local FX graph cache feature, i.e., by storing artifacts on the host’s temp directory. ``1`` enables, and any other value disables it. By default, the disk location is per username, but users can enable sharing across usernames by specifying ``TORCHINDUCTOR_CACHE_DIR`` (below).
```
Comment (masnesral) [inline recipes_source/torch_compile_caching_tutorial.rst:33]: on -> in

File: recipes_source/torch_compile_caching_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+TORCHINDUCTOR_CACHE_DIR
+~~~~~~~~~~~~~~~~~~~~~~~~
+This setting specifies the location of all on-disk caches. By default, the location is in the system temp directory under ``torchinductor_<username>``, for example, ``/tmp/torchinductor_myusername``.
+
+Note that if ``TRITON_CACHE_DIR`` is not set in the environment, Inductor sets the Triton cache directory to this same temp location, under the Triton subdirectory.
+
+TORCHINDUCTOR_FX_GRAPH_REMOTE_CACHE
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This setting enables the remote FX graph cache feature. The current implementation uses Redis. ``1`` enables caching, and any other value disables. The following environment variables configure the host and port of the Redis server:
```
Comment (masnesral) [inline recipes_source/torch_compile_caching_tutorial.rst:43]: disables it

**Review by masnesral — APPROVED** [review]

**Review by masnesral — APPROVED** [review]

File: recipes_source/torch_compile_caching_tutorial.rst (hunk 1 above)
Comment (svekars) [reply recipes_source/torch_compile_caching_tutorial.rst]: You could say something like: These caching mechanisms operate seamlessly, requiring no user intervention.

**Pull request merged**